hs-abci-server/src/Network/ABCI/Types/Messages/Request.hs

```diff
@@ -56,6 +62,22 @@ data Request (m :: MessageType) :: * where
   RequestEndBlock :: EndBlock -> Request 'MTEndBlock
   RequestCommit :: Commit -> Request 'MTCommit
 
+instance ToJSON (Request (t :: MessageType)) where
+  toJSON (RequestEcho v)       = toJSON v
+  toJSON (RequestFlush v)      = toJSON v
+  toJSON (RequestInfo v)       = toJSON v
+  toJSON (RequestSetOption v)  = toJSON v
+  toJSON (RequestInitChain v)  = toJSON v
+  toJSON (RequestQuery v)      = toJSON v
+  toJSON (RequestBeginBlock v) = toJSON v
+  toJSON (RequestCheckTx v)    = toJSON v
+  toJSON (RequestDeliverTx v)  = toJSON v
+  toJSON (RequestEndBlock v)   = toJSON v
+  toJSON (RequestCommit v)     = toJSON v
+
+instance FromJSON (Request (t :: MessageType)) where
```

Comment:
we can remove this as there is currently no intended use for these instances. In general you can't write instances lke this anyway, you would need to write them individually like

```haskell
instance FromJSON (Request 'MTEcho) where
...
```